gui/designwidget.cc

```diff
@@ -265,11 +265,49 @@ void DesignWidget::newContext(Context *ctx)
         std::lock_guard<std::mutex> lock_ui(ctx->ui_mutex);
         std::lock_guard<std::mutex> lock(ctx->mutex);
         
-        getTreeByElementType(ElementType::BEL)->loadContext(ElementType::BEL, ctx);
-        getTreeByElementType(ElementType::WIRE)->loadContext(ElementType::WIRE, ctx);
-        getTreeByElementType(ElementType::PIP)->loadContext(ElementType::PIP, ctx);
-        getTreeByElementType(ElementType::CELL)->loadContext(ElementType::CELL, ctx);
-        getTreeByElementType(ElementType::NET)->loadContext(ElementType::NET, ctx);
+        {
+            std::map<std::pair<int, int>, std::vector<BelId>> belMap;

```

Comment:
I think we could use a typedef for the pair here. I don't immediately follow this code.

Reply:
It doesn't even have to be a global scope typedef.

Reply:
yeah will use proper typedef for this